### ISSUE TITLE
Duplicate Key insert in internal block.

### ISSUE THREAD

**Issue by timtadh** [timtadh/fs2#2]
```
2015/06/25 14:32:52 would have inserted a duplicate key, [0 0 0 40]
goroutine 5 [running]:
github.com/timtadh/fs2/errors.Errorf(0xc20e8272c0, 0x2f, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/errors/error.go:16 +0x92
github.com/timtadh/fs2/bptree.(*internal).putKey(0x7fded5867000, 0x0, 0x7fded63de020, 0x4, 0xfe0, 0xc20ee88ea8, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/internal.go:277 +0x3c4
github.com/timtadh/fs2/bptree.(*internal).putKP(0x7fded5867000, 0x0, 0x7fded63de020, 0x4, 0xfe0, 0xbe2000, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/internal.go:211 +0x1d8
github.com/timtadh/fs2/bptree.func·073(0x7fded63de020, 0x4, 0xfe0, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/insert.go:156 +0x18b
github.com/timtadh/fs2/bptree.func·023(0x7fded63de000, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/first.go:20 +0x124
github.com/timtadh/fs2/bptree.func·021(0x7fded63de000, 0x1000, 0x1000, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/do.go:103 +0xfa
github.com/timtadh/fs2/fmap.(*BlockFile).Do(0xc20cc28240, 0xbe2000, 0x1, 0xc20ee890a8, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/fmap/fmap.go:559 +0xbf
github.com/timtadh/fs2/bptree.(*BpTree).do(0xc20fc78fc0, 0xbe2000, 0xc20ee89108, 0xc20ee890f8, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/do.go:107 +0x79
github.com/timtadh/fs2/bptree.(*BpTree).firstKey(0xc20fc78fc0, 0xbe2000, 0xc20ee891b8, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/first.go:22 +0x84
github.com/timtadh/fs2/bptree.func·074(0x7fded5867000, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/insert.go:157 +0x253
github.com/timtadh/fs2/bptree.func·021(0x7fded5867000, 0x1000, 0x1000, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/do.go:101 +0xab
github.com/timtadh/fs2/fmap.(*BlockFile).Do(0xc20cc28240, 0x6b000, 0x1, 0xc20ee892e0, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/fmap/fmap.go:559 +0xbf
github.com/timtadh/fs2/bptree.(*BpTree).do(0xc20fc78fc0, 0x6b000, 0xc20ee89400, 0x685ea8, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/do.go:107 +0x79
github.com/timtadh/fs2/bptree.(*BpTree).doInternal(0xc20fc78fc0, 0x6b000, 0xc20ee89400, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/do.go:43 +0x5a
github.com/timtadh/fs2/bptree.(*BpTree).internalInsert(0xc20fc78fc0, 0x6b000, 0xc209b61dc0, 0x4, 0x4, 0xc209b61dc8, 0x8, 0x8, 0x18, 0xc209b61dc8, ...)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/insert.go:160 +0x273
github.com/timtadh/fs2/bptree.(*BpTree).insert(0xc20fc78fc0, 0x6b000, 0xc209b61dc0, 0x4, 0x4, 0xc209b61dc8, 0x8, 0x8, 0x0, 0xc209b61dc8, ...)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/insert.go:100 +0x13c
github.com/timtadh/fs2/bptree.(*BpTree).Add(0xc20fc78fc0, 0xc209b61dc0, 0x4, 0x4, 0xc209b61dc8, 0x8, 0x8, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/insert.go:54 +0x2ff
github.com/timtadh/fs2/bptree.(*Varchar).szAdd(0xc20f688390, 0x24, 0xa24c34, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/varchar.go:407 +0xd2
github.com/timtadh/fs2/bptree.(*Varchar).free(0xc20f688390, 0xa24c34, 0x24, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/varchar.go:397 +0x5f2
github.com/timtadh/fs2/bptree.(*Varchar).freeExtra(0xc20f688390, 0xa24bc8, 0x90, 0x6c, 0x0, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/varchar.go:268 +0xfd
github.com/timtadh/fs2/bptree.(*Varchar).newRun(0xc20f688390, 0xa24bc8, 0x5c, 0x6c, 0x90, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/varchar.go:675 +0x13a
github.com/timtadh/fs2/bptree.(*Varchar).Alloc(0xc20f688390, 0x5c, 0xa24bc8, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/varchar.go:222 +0x479
github.com/timtadh/fs2/bptree.(*BpTree).checkValue(0xc20fc78fe0, 0xc20e6f8f00, 0x5c, 0x5c, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/insert.go:15 +0xa8
github.com/timtadh/fs2/bptree.(*BpTree).Add(0xc20fc78fe0, 0xc20e6f8d20, 0x5c, 0x5c, 0xc20e6f8f00, 0x5c, 0x5c, 0x0, 0x0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fs2/bptree/insert.go:50 +0x230
github.com/timtadh/fsm/store.(*Fs2BpTree).Add(0xc20fc79020, 0xc20e6f8d20, 0x5c, 0x5c, 0xc20bfc5f00)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fsm/store/fs.go:146 +0x769
github.com/timtadh/fsm/mine.(*Queue).Add(0xc20cc282c0, 0xc20ba58c00)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fsm/mine/queue.go:66 +0x1d1
github.com/timtadh/fsm/mine.func·014(0xc20ba58c00)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fsm/mine/depth.go:123 +0x79
github.com/timtadh/fsm/mine.(*DepthMiner).process(0xc20e6a41c0, 0xc20c0f1020, 0xc20ee89f18)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fsm/mine/depth.go:181 +0x440
github.com/timtadh/fsm/mine.(*DepthMiner).search(0xc20e6a41c0, 0xa)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fsm/mine/depth.go:128 +0x59f
github.com/timtadh/fsm/mine.(*DepthMiner).mine(0xc20e6a41c0)
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fsm/mine/depth.go:79 +0x31
created by github.com/timtadh/fsm/mine.Depth
        /home/hendersont/stuff/research/fsm/src/github.com/timtadh/fsm/mine/depth.go:73 +0x321
Command exited with non-zero status 1
```

**Comment by timtadh** [comment]
I think I found the issue that was causing this. Testing the fix.

**Comment by timtadh** [comment]
should be fixed in b9640f14d4d37378fadc7558b0d1bcf16bc47f76
